webapp/src/components/properties/dateRange/dateRange.tsx

```diff
@@ -0,0 +1,250 @@
+            if (singleDate && DateUtils.isDate(singleDate)) {
+                dateProperty.from = singleDate.getTime()
+            } else {
+                dateProperty = JSON.parse(initialValue)
+            }
+        }
+        return dateProperty
+    }
+
+    const [dateProperty, setDateProperty] = useState<DateProperty>(createDatePropertyFromString(value as string))
+    const [showDialog, setShowDialog] = useState(false)
+
+    // Keep dateProperty as UTC,
+    // dateFrom / dateTo will need converted to local time, to ensure date stays consistent
+    // dateFrom / dateTo will be used for input and calendar dates
+    const dateFrom = dateProperty.from ? new Date(dateProperty.from + (dateProperty.includeTime ? 0 : timeZoneOffset)) : undefined
+    const dateTo = dateProperty.to ? new Date(dateProperty.to + (dateProperty.includeTime ? 0 : timeZoneOffset)) : undefined
+    const [fromInput, setFromInput] = useState<string>(getDisplayDate(dateFrom))
+    const [toInput, setToInput] = useState<string>(getDisplayDate(dateTo))
+
+    const isRange = dateTo !== undefined
+
+    const locale = intl.locale.toLowerCase()
+    if (locale && locale !== 'en' && !loadedLocales[locale]) {
+        /* eslint-disable global-require */
+        loadedLocales[locale] = require(`moment/locale/${locale}`)
+        /* eslint-disable global-require */
+    }
+
+    const handleDayClick = (day: Date) => {
+        const range : DateProperty = {}
+        if (isRange) {
+            const newRange = DateUtils.addDayToRange(day, {from: dateFrom, to: dateTo})
+            range.from = newRange.from?.getTime()
+            range.to = newRange.to?.getTime()
+        } else {
+            range.from = day.getTime()
+            range.to = undefined
+        }
+        saveRangeValue(range)
+    }
+
+    const onRangeClick = () => {
+        let range : DateProperty = {
+            from: dateFrom?.getTime(),
+            to: dateFrom?.getTime(),
+        }
+        if (isRange) {
+            range = ({
+                from: dateFrom?.getTime(),
+                to: undefined,
+            })
+        }
+
+        saveRangeValue(range)
+    }
+
+    const onClear = () => {
+        saveRangeValue({})
+    }
+
+    const saveRangeValue = (range: DateProperty) => {
+        const rangeUTC = {...range}
+        if (rangeUTC.from) {
+            rangeUTC.from -= dateProperty.includeTime ? 0 : timeZoneOffset
+        }
+        if (rangeUTC.to) {
+            rangeUTC.to -= dateProperty.includeTime ? 0 : timeZoneOffset
+        }
+
+        setDateProperty(rangeUTC)
+        setFromInput(getDisplayDate(range.from ? new Date(range.from) : undefined))
+        setToInput(getDisplayDate(range.to ? new Date(range.to) : undefined))
+    }
+
+    let displayValue = ''
+    if (dateFrom) {
+        displayValue = getDisplayDate(dateFrom)
+    }
+    if (dateTo) {
+        displayValue += ' -> ' + getDisplayDate(dateTo)
+    }
+
+    const onClose = () => {
```

Comment:
@jespino I was not able to get this to work using `useCallback`. This [article](https://stackoverflow.com/questions/57847594/react-hooks-accessing-up-to-date-state-from-within-a-callback) discusses a couple different options. Previously the `useRef`option was used, changed this to retrieve the current state, via `setDateProperty`.